### PR TITLE
`/#contact` linking fixed 

### PR DESCRIPTION
**Pull request by mwilkins74**
Got this linking fixed when the Contact `FooterLink` is clicked.

Stuck on how to get the same functionality when the Contact Link is clicked inside of the `NavList`.

### CODE REVIEW

File: src/components/templates/Home.tsx [hunk 1]
```diff
@@ -13,7 +13,7 @@ export function Home() {
     <>
       <LandingPageLinks pages={[...recentWorkPageInfo.orderedPages]} />
       <VideoList videos={videos} pageType="recent" />
-      <Contact />
+      <Contact id="contact" />
```
Comment (ChuckBTaylor) [inline src/components/templates/Home.tsx:16]: ❓ Does this need to be a prop?  Since the component is called `Contact`, would it ever be a different value?

Comment (kmdunn5) [reply src/components/templates/Home.tsx:16]: The only reason I can see this wanting to be a prop is if we ever render two contact components. But I don't think we do, so I think it makes sense to keep it inside the component.

**Comment by kmdunn5** [comment]
Oh, @mwilkins74 I remember having this issue when I first added this before stuff got changed. The `/#contact` only works with a `<Link>` component. It doesn't work with the `navigate("/#contact")`. I had to do [this](https://github.com/mwkwsd/sensensomething/blob/9e36a86fe4999f065ab8fa4c56545798af282a2f/src/components/molecules/About/AboutPage.tsx#L16) to make it work for the button. The `to="/#contact"` is necessary.

**Comment by mwilkins74** [comment]
Oh fantastic!
That's what I _was_ able to determine - that it only worked with the '<Link', but my brain didn't even think to implement to '"to="' attribute. Duh!

**Comment by mwilkins74** [comment]
Ok, I removed the prop from the Home component and kept it inside of the Contact component. 
The Contact `FooterLink` and the `Like My Work` button works to jump to the Contact component. 
In the NavList component, it won't jump directly to the Contact component. When I click the Contact link inside of the NavList the route changes to`http://localhost:3000/#contact`, but it just renders the top of the Home component. 
I'm not seeing what I'm missing?

**Comment by ChuckBTaylor** [comment]
I think you've got your cause-effect wrong.  None of the links are working properly. The Footer and Like My Work? Link just look like they're working correctly

File: src/components/atoms/utilityComponents/ScrollToHashElement.tsx
Comment (ChuckBTaylor) [inline src/components/atoms/utilityComponents/ScrollToHashElement.tsx:1]: Utility component to scroll around the application

File: src/components/atoms/utilityComponents/ScrollToHashElement.tsx [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+export function ScrollToHasElement(): React.ReactNode {
+  const location = useLocation()
+  const lastHash = useRef('')
+
+  useEffect(() => {
+    if (location.hash) {
+      lastHash.current = location.hash.slice(1)
+    } else {
+      window.scrollTo(0, 0)
```
Comment (ChuckBTaylor) [inline src/components/atoms/utilityComponents/ScrollToHashElement.tsx:17]: react-router-dom does not by default scroll to the top of the page when changing routes.

File: src/components/templates/NavBar.tsx [hunk 3]
```diff
@@ -45,6 +44,7 @@ export function NavBar() {
       </AppBar>
 
       <Drawer
+        disableRestoreFocus
```
Comment (ChuckBTaylor) [inline src/components/templates/NavBar.tsx:47]: Prevents page from scrolling back to top after link is clicked

**Review by ChuckBTaylor — APPROVED** [review]
@mwilkins74 I'm approving, but not merging until you look through it and approve my changes as well.

**Comment by mwilkins74** [comment]
> I think you've got your cause-effect wrong. None of the links are working properly. The Footer and "Like My Work?" link just _look_ like they're working correctly

Can you elaborate a bit for me, please?
When clicked the links are jumping to the Contact form so how are they not working? I don't understand.

File: src/components/molecules/NavList.tsx [hunk 4]
```diff
@@ -16,13 +16,20 @@ const orderedLinks: (Genre | Role)[] = [
...
   const navigate = useNavigate()
 
-  const secondNavLinks = orderedLinks.map(linkEnum => (
-    <NavLinks
-      key={`${linkEnum}`}
-      label={enumMappings[linkEnum].label}
-      onClick={() => navigate(enumMappings[linkEnum].url)}
-    />
-  ))
+  const secondNavLinks = orderedLinks.map(linkEnum => {
```
Comment (mwilkins74) [inline src/components/molecules/NavList.tsx]: I like this so much better.

**Comment by mwilkins74** [comment]
All of your changes look great. 
Really like the Utility file to scroll to the hash. I never would have thought of that.

**Pull request merged**